lib/preprocessor.js

```diff
@@ -51,64 +45,69 @@ var createPreprocessor = function (config, basePath, injector) {
 
   return function preprocess (file, done) {
     patterns = Object.keys(config)
-    var thisFileIsBinary = isBinary[path.extname(file.originalPath).toLowerCase()]
-    var preprocessors = []
-
-    var nextPreprocessor = function (error, content) {
-      // normalize B-C
-      if (arguments.length === 1 && typeof error === 'string') {
-        content = error
-        error = null
+    isBinaryFile(file.originalPath, function (err, thisFileIsBinary) {
+      if (err) {
+        throw err
       }
 
-      if (error) {
-        file.content = null
-        file.contentPath = null
-        return done(error)
-      }
+      var preprocessors = []
 
-      if (!preprocessors.length) {
-        file.contentPath = null
-        file.content = content
-        file.sha = sha1(content)
-        return done()
+      var nextPreprocessor = function (error, content) {
```

Comment:
I factored out `nextPreprocessor` into it's own function.  I had to change the signature, which I hope is okay.  Otherwise it would have to be a factor function.